xds/internal/client/tests/client_test.go

```diff
@@ -42,77 +43,67 @@ const (
 	testXDSServer = "xds-server"
 )
 
-func clientOpts(balancerName string) xdsclient.Options {
-	return xdsclient.Options{
-		Config: bootstrap.Config{
-			BalancerName: balancerName,
-			Creds:        grpc.WithInsecure(),
-			NodeProto:    testutils.EmptyNodeProtoV2,
-		},
-	}
-}
-
 func (s) TestNew(t *testing.T) {
 	tests := []struct {
 		name    string
-		opts    xdsclient.Options
+		config  *bootstrap.Config
 		wantErr bool
 	}{
-		{name: "empty-opts", opts: xdsclient.Options{}, wantErr: true},
+		{
+			name:    "empty-opts",
+			config:  &bootstrap.Config{},
+			wantErr: true,
+		},
 		{
 			name: "empty-balancer-name",
-			opts: xdsclient.Options{
-				Config: bootstrap.Config{
-					Creds:     grpc.WithInsecure(),
-					NodeProto: testutils.EmptyNodeProtoV2,
-				},
+			config: &bootstrap.Config{
+				Creds:     grpc.WithInsecure(),
+				NodeProto: testutils.EmptyNodeProtoV2,
 			},
 			wantErr: true,
 		},
 		{
 			name: "empty-dial-creds",
-			opts: xdsclient.Options{
-				Config: bootstrap.Config{
-					BalancerName: testXDSServer,
-					NodeProto:    testutils.EmptyNodeProtoV2,
-				},
+			config: &bootstrap.Config{
+				BalancerName: testXDSServer,
+				NodeProto:    testutils.EmptyNodeProtoV2,
 			},
 			wantErr: true,
 		},
 		{
 			name: "empty-node-proto",
-			opts: xdsclient.Options{
-				Config: bootstrap.Config{
-					BalancerName: testXDSServer,
-					Creds:        grpc.WithInsecure(),
-				},
+			config: &bootstrap.Config{
+				BalancerName: testXDSServer,
+				Creds:        grpc.WithInsecure(),
 			},
 			wantErr: true,
 		},
 		{
 			name: "node-proto-version-mismatch",
-			opts: xdsclient.Options{
-				Config: bootstrap.Config{
-					BalancerName: testXDSServer,
-					Creds:        grpc.WithInsecure(),
-					NodeProto:    testutils.EmptyNodeProtoV3,
-					TransportAPI: version.TransportV2,
-				},
+			config: &bootstrap.Config{
+				BalancerName: testXDSServer,
+				Creds:        grpc.WithInsecure(),
+				NodeProto:    testutils.EmptyNodeProtoV3,
+				TransportAPI: version.TransportV2,
 			},
 			wantErr: true,
 		},
 		// TODO(easwars): Add cases for v3 API client.
 		{
 			name: "happy-case",
-			opts: clientOpts(testXDSServer),
+			config: &bootstrap.Config{
+				BalancerName: testXDSServer,
+				Creds:        grpc.WithInsecure(),
+				NodeProto:    testutils.EmptyNodeProtoV2,
+			},
 		},
 	}
 
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
-			c, err := xdsclient.New(test.opts)
+			c, err := xdsclient.NewWithConfigForTesting(test.config, 15*time.Second)
```

Comment:
I removed the zero check from `newWithConfig` though. Since this is now a separate parameter (instead of a field of a struct), all caller should be aware and set a value.